GitUI/CommandsDialogs/FormDeleteRemoteBranch.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Forms;
+using GitCommands;
+using ResourceManager;
+using GitUIPluginInterfaces;
+using System.IO;
+using GitUI.Script;
+
+namespace GitUI.CommandsDialogs
+{
+    public sealed partial class FormDeleteRemoteBranch : GitModuleForm
+    {
+        private readonly TranslationString _deleteRemoteBranchesCaption = new TranslationString("Delete remote branches");
+        private readonly TranslationString _confirmDeleteUnmergedRemoteBranchMessage =
+            new TranslationString("At least one remote branch is unmerged. Are you sure you want to delete it?" + Environment.NewLine + "Deleting a branch can cause commits to be deleted too!");
+
+        private readonly string _defaultRemoteBranch;
+        private readonly HashSet<string> _mergedBranches = new HashSet<string>();
+
+        public FormDeleteRemoteBranch(GitUICommands aCommands, string defaultRemoteBranch)
+            : base(aCommands)
+        {
+            InitializeComponent();
+            Translate();
+            _defaultRemoteBranch = defaultRemoteBranch;
+        }
+
+        private void FormDeleteRemoteBranchLoad(object sender, EventArgs e)
+        {
+            Branches.BranchesToSelect = Module.GetRefs(tags: true, branches: true).Where(h => h.IsRemote).ToList();
+            foreach (var branch in Module.GetMergedRemoteBranches())
+            {
+                _mergedBranches.Add(branch);
+            }
+
+            if (_defaultRemoteBranch != null)
+            {
+                Branches.SetSelectedText(_defaultRemoteBranch);
+            }
+        }
+
+        private void OkClick(object sender, EventArgs e)
+        {
+            try
+            {
+                if (!DeleteRemote.Checked)
+                {
+                    return;
+                }
+
+                List<IGitRef> selectedBranches = Branches.GetSelectedBranches().ToList();
+
+                var hasUnmergedBranches = selectedBranches.Any(branch => !_mergedBranches.Contains(branch.CompleteName));
+
+                if (hasUnmergedBranches)
+                {
+                    if (MessageBox.Show(this, _confirmDeleteUnmergedRemoteBranchMessage.Text, _deleteRemoteBranchesCaption.Text, MessageBoxButtons.YesNo) != DialogResult.Yes)
```

Comment:
I think it only makes sense to show this message if the remote doesn't have any reachable commits on top of it
For example, in this situation the message this warning is not appropriate:

![image](https://user-images.githubusercontent.com/4403806/33739134-0fbe631e-dbf0-11e7-8712-0303f0692ffa.png)

Thoughts?

Reply:
I am not sure if there is a reliable way to detect that for remote branches. For local ones it's all clear, you have all the data, for remote ones you might not.